apps/frontend/src/app/PageSettings/DatabaseCard.tsx

```diff
@@ -82,6 +250,30 @@ function DataCard({ index, readOnly }: { index: number; readOnly: boolean }) {
     database.toExtraLocalDB()
   }, [database, name])
 
+  const onDriveUpload = useCallback(() => {
```

Comment:
I think we should maintain 2 states `isUploading` & `isDownloading` that prevents the button/operation from being clicked while downloading/uploading. Can leverage a `callback` in `uploadDataToDrive` and `downloadDataFromDrive` to enable the button once the operation is done.

Reply:
Yeah I agree,
I'll look into how react state management works and will try to implement it this weekend

Reply:
I added CloudSyncContext, it works but I'm not sure if it's compliant with best practices, any feedback appreciated